### PR TITLE
OKTA-577531 Auth JS redirect guide

### PR DESCRIPTION
**Pull request by brianduffield-okta**
## Description:
- **What's changed?** Initial draft of a redirect guide using Auth JS.
- **Is this PR related to a Monolith release?** n/a

### Resolves:

* [OKTA-577531](https://oktainc.atlassian.net/browse/OKTA-5775331)

### CODE REVIEW

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx Scorecards

Acrolinx checked the content changed here.

Please review the scorecards for each file and make the recommended
improvements.

| Article | Score | Issues | Scorecard | Status |
| ------- | ----- | ------ | --------- | ------ |
| [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/index.md](https://github.com/okta/okta-developer-docs/blob/6ba2f2b32074d2c73945099b4ba78c8c67b5ab66/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Findex.md) | 100 | 6 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/1a17875e-8436-43f2-adcf-dd9727e18ba6) | [:white_check_mark:](# "Successfully processed") |
| [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md](https://github.com/okta/okta-developer-docs/blob/6ba2f2b32074d2c73945099b4ba78c8c67b5ab66/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Fmain%2Findex.md) | 82 | 45 | [→](https://okta.acrolinx.cloud/api/v1/checking/scorecards/12dbe6cb-6b49-41e9-a4c4-d573bbb2e5ac) | [:white_check_mark:](# "Successfully processed") |

Successfully checked 2 of 2 documents.
See summary in [Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.dcba0928-4fbd-4e2f-be22-245f2f6b01fd?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

Reopen the pull request or push new changes to check again.

Depending on the Acrolinx server configuration, the
links expire after some time and you must have a login for the
Acrolinx server to access them again.

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/index.md](https://github.com/okta/okta-developer-docs/blob/33488a580ca21c265b8af08c0e2ebc614cbaf4f5/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Findex.md) | [100](https://okta.acrolinx.cloud/api/v1/checking/scorecards/fb89e2d8-650a-4dc7-b304-7e1ad177ade1) | 100  | 100  | 100  | 100  |
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md](https://github.com/okta/okta-developer-docs/blob/33488a580ca21c265b8af08c0e2ebc614cbaf4f5/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Fmain%2Findex.md) | [81](https://okta.acrolinx.cloud/api/v1/checking/scorecards/1254f336-0614-4c4c-bce6-6232daf38063) | 75  | 94  | 46  | 100  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.61829100-1ee2-45be-8a95-12a412cbf87e?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

**Comment by acrolinxdevdocs-okta** [comment]
## Acrolinx score **DRAFT 06.01.22**
 
A minimum Acrolinx Score of **40** is required. The total score is an average of the subscores.

Select **Total score** to review the Acrolinx scorecard for your article. Try to increase your individual scores, ex. Correctness. Your article will be clearer and more consistent with Microsoft standards.
 
| Article | **Total score**<br />**Required:70** | Word and phrases<br />(Brand, terms)<br />Preferred: 40 | Correctness<br />(Spelling, grammar)<br />Preferred: 40 | Clarity<br />(Readability)<br />Preferred: 10 | Inclusive language<br />(+ accesibility)<br />Preferred: 10 |
|---------|:------------------------------------:|:--------------------------------------------------------:|:--------------------------------------------------------:|:---------------------------------------------:|:------------------------------------------------------------:|
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/index.md](https://github.com/okta/okta-developer-docs/blob/f179ae4104af892ac92c903c4b286d17fd8a4daf/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Findex.md) | [100](https://okta.acrolinx.cloud/api/v1/checking/scorecards/21b04190-52e2-4fcf-817e-442ffe51f28b) | 100  | 100  | 100  | 100  |
| [:white_check_mark:](# "Successfully processed") [packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md](https://github.com/okta/okta-developer-docs/blob/f179ae4104af892ac92c903c4b286d17fd8a4daf/packages%2F%40okta%2Fvuepress-site%2Fdocs%2Fguides%2Fauth-js-redirect%2Fmain%2Findex.md) | [81](https://okta.acrolinx.cloud/api/v1/checking/scorecards/5d62ba77-18d9-435e-b980-0c298508f868) | 75  | 94  | 46  | 100  |

  - The results from this response were procced by the [hugo-targets1.solution.acrolinx.sh](https://hugo-targets1.solution.acrolinx.sh) Acrolinx Cloud Host.
  - Scorecard links by default are only good for 7 days. [Link for this commit's Content Analysis Dashboard](https://okta.acrolinx.cloud/api/batch/gen.github.599d322e-3299-4f00-947f-2a36d3b98ac4?csBaseUrl=https%3A%2F%2Fokta.acrolinx.cloud&X-Acrolinx-Client-Locale=en)

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
+---
+title: Sign users in to your SPA using the redirect model and Auth JS
+---
+
+The [Okta JavaScript Auth SDK](https://github.com/okta/okta-auth-js), (Auth JS), provides functionality to help implement a wide variety of web authentication solutions for both the [redirect and embedded model](/docs/concepts/redirect-vs-embedded/). This guide shows you how to create a simple redirect authentication solution using Auth JS, which can be dropped into just about any front-end or server-side web application.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:5]:
```suggestion
The [Okta JavaScript Auth SDK](https://github.com/okta/okta-auth-js), (Auth JS), provides functionality to help implement a wide variety of web authentication solutions for both the [redirect and embedded model](/docs/concepts/redirect-vs-embedded/). This guide shows you how to create a simple redirect authentication solution using Auth JS, which you can drop into just about any front-end or server-side web application.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+* [Okta Developer Edition organization](https://developer.okta.com/signup/)
+
+---
+
+## About the Okta Auth JavaScript SDK
+
+The Okta Auth JS SDK builds on top of the [Authentication API](/docs/reference/api/authn/), the [OpenID Connect API](/docs/reference/api/oidc/), and the Identity Engine [interaction code](/docs/concepts/interaction-code/) flow. This SDK enables you to create various sign-in experiences.
+
+These experiences include fully branded embedded authentication, as with [Auth JS fundamentals](/docs/guides/auth-js/), and redirect authentication. Auth JS is used by the Okta [Sign-In Widget](https://github.com/okta/okta-signin-widget), which powers the default Okta sign-in page. It also powers our other redirect SDKs that provide simple authentication for server-side web apps and single-page JavaScript apps (SPA). See the [Quickstart guides](/docs/guides/quickstart/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:24]:
```suggestion
These experiences include fully branded embedded authentication, as with [Auth JS fundamentals](/docs/guides/auth-js/) and redirect authentication. Auth JS is used by the Okta [Sign-In Widget](https://github.com/okta/okta-signin-widget), which powers the default Okta sign-in page. It also powers our other redirect SDKs that provide simple authentication for server-side web apps and single-page JavaScript apps (SPA). See the [Quickstart guides](/docs/guides/quickstart/).
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+In this guide, you don’t need to use a specific server-side or front-end framework that Okta officially supports to get access to redirect authentication. It's possible to use Auth JS to create a drop-in solution that works with most web apps, whether you're adding centralized sign-in to a new app or retrofitting it to an existing app. To see examples of Auth JS with other front-end frameworks, go to [Sign in to SPA](/docs/guides/sign-into-spa-redirect/angular/main/).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:26]:
```suggestion
In this guide, you don’t need to use a specific server-side or front-end framework that Okta officially supports to get access to redirect authentication. It's possible to use Auth JS to create a drop-in solution that works with most web apps, whether you're adding a centralized sign-in flow to a new app or retrofitting it to an existing app. To see examples of Auth JS with other front-end frameworks, go to [Sign in to SPA](/docs/guides/sign-into-spa-redirect/angular/main/).
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+
+If you'd like to explore the entire Auth JS SDK, see [Okta Auth JS JavaScript SDK](https://github.com/okta/okta-auth-js/#readme).
+
+## Create an Okta app integration
+
+An Okta app integration represents your app in your Okta org. The integration configures how your app integrates with the Okta services including which users and groups have access, authentication policies, token refresh requirements, redirect URLs, and more. The integration includes configuration information required by the app to access Okta.
+
+1. [Sign in to your Okta organization](https://developer.okta.com/login) with your administrator account.
+1. Click the **Admin** button on the top right of the page.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:35]:
```suggestion
1. Click **Admin** in the upper-right corner of the page.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+1. Open the Applications configuration pane by selecting **Applications** > **Applications**.
+1. Click **Create App Integration**.
+1. Select a **Sign-in method** of **OIDC - OpenID Connect**, then click **Next**.
+1. Select an **Application type** of **Single-Page Application**, then click **Next**.
+   > **Note:** If you choose an inappropriate application type, it breaks the sign-in or sign-out flows by requiring the verification of a client secret, which public clients don't have.
+1. Enter an **App integration name**.
+1. Select the **Authorization Code** grant type.
+1. Enter the **Sign-in redirect URIs** for local development. For this sample, use `http://localhost:9000`.
+1. Enter the **Sign-out redirect URIs** for local development. For this sample, use `http://localhost:9000`. For more information on callback URIs, see [Define callback route](#define-a-callback-route).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:44]:
```suggestion
1. Enter the **Sign-out redirect URIs** for local development. For this sample, use `http://localhost:9000`. See [Define callback route](#define-a-callback-route).
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+1. In the **Assignments** section, define the type of **Controlled access** for your app. Select **Allow everyone in your organization to access**. For more information, see the [Assign app integrations](https://help.okta.com/okta_help.htm?type=oie&id=ext-lcm-user-app-assign) topic in the Okta product documentation.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:45]:
```suggestion
1. In the **Assignments** section, define the type of **Controlled access** for your app. Select **Allow everyone in your organization to access**. See [Assign app integrations](https://help.okta.com/okta_help.htm?type=oie&id=ext-lcm-user-app-assign).
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+1. Make sure to clear the checkbox **Enable immediate access with Federation Broker Mode**.
+1. Click **Save** to create the app integration. The configuration pane for the integration opens after it's saved. Keep this pane open as you'll need to copy the **Client ID** and your org domain name when configuring your app.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:47]:
```suggestion
1. Click **Save** to create the app integration. The configuration pane for the integration opens after it's saved. Keep this pane open as you need to copy the **Client ID** and your org domain name when configuring your app.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+### Enable Trusted Origins
+
+Reduce possible attack vectors by defining Trusted Origins, which are the websites allowed to access the Okta API for your app integration. Cross-Origin Resource Sharing (CORS) enables JavaScript requests using `XMLHttpRequest` with the Okta session cookie. For information on enabling CORS, see [Grant cross-origin access to websites](/docs/guides/enable-cors/main/#grant-cross-origin-access-to-websites).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:51]:
```suggestion
Reduce possible attack vectors by defining Trusted Origins, which are the websites allowed to access the Okta API for your app integration. Cross-Origin Resource Sharing (CORS) enables JavaScript requests using `XMLHttpRequest` with the Okta session cookie. See [Grant cross-origin access to websites](/docs/guides/enable-cors/main/#grant-cross-origin-access-to-websites).
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+
+>**Note:** To reduce risk, only grant access to the Okta API to specific websites (origins) that you both control and trust.
+
+To set trusted origins go to **Security** > **API** and select the **Trusted Origins** tab. See [Enable Trusted Origins](/docs/guides/enable-cors/). Add the URL `http://localhost:9000`.
+
+## Create a basic app
+
+To make this sample as versatile as possible, the following starter app redirects to Okta to sign in as you load it into the browser. In your own apps, you might want to initiate the redirect through the press of a sign-in button, or when visiting a certain route that requires authentication (such as an admin page). The key is that you initiate the sign in through the redirect.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:59]:
```suggestion
To make this sample as versatile as possible, the following starter app redirects to Okta to sign in as you load it into the browser. In your own apps, you might want to initiate the redirect through the press of a sign-in button, or when visiting a certain route that requires authentication (such as an admin page). The key is that you initiate the sign in flow through the redirect.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+  clientId: appClientID,
+  redirectUri: "http://localhost:9000/", //or the redirect URI for your app
+  issuer: baseOktaURL , // oidc
+  scopes: ['openid', 'profile', 'email']
+});
+
+</script>
+```
+
+This code initializes the SDK by creating an instance of the `OktaAuth()` object. The object stores all the necessary config information for your auth session, and is used to control subsequent steps of the process.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:120]:
```suggestion
This code initializes the SDK by creating an instance of the `OktaAuth()` object. The object stores all the necessary config information for your auth session and is used to control subsequent steps of the process.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+
+Go to `http://localhost:9000`. The sample app starts the redirect flow when the page opens.
+
+### Troubleshoot your app
+
+If your app isn't working, ensure that:
+
+* Your org URL is accurate and formatted correctly, including the secure protocol, `https://`.
+* Your client ID is accurate from your Okta app integration.
+* Your `redirectUri` is accurate or the web-server port number is correct.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:179]:
```suggestion
* Your `redirectUri` is accurate or the web server port number is correct.
```

File: packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+* You've enabled a Trusted Origin for `http://localhost:9000`. See [Enable Trusted Origin](#enable-trusted-origins).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/auth-js-redirect/main/index.md:180]:
```suggestion
* You've enabled a Trusted Origin for `http://localhost:9000`. See [Enable Trusted Origins](#enable-trusted-origins).
```

**Review by susanharper-okta — APPROVED** [review]
some nits - approving to not hold you up

**Comment by danmarma-okta** [comment]
This looks great, okay to publish. Cheers!

**Pull request merged**